webpack.config.js

```diff
@@ -379,6 +380,112 @@ var config = {
     ],
 };
 
+async function initializeModuleFederation() {
+    function makeSingletonSharedModules(packageNames) {
+        const sharedObject = {};
+
+        for (const packageName of packageNames) {
+            const version = packageJson.dependencies[packageName];
+
+            sharedObject[packageName] = {
+                requiredVersion: version,
```

Comment:
I haven't quite figured out why, but we have to specify an explicit version and required version for singleton shared modules. If we just pass in a string (like with `luxon` and `@mattermost/client` below), it determines it automatically from the package.json

Reply:
Great find that the web app is using Focalboard's version of React. It looks like the way those fields work, `requiredVersion` enforces which version of the module will be used by this part of the code while `version` is used by Webpack to figure out which container's module to load.

Since we were specifying a higher `version` of React in Boards, it was causing Webpack to load React 16.14.0 instead of 16.13.1 which caused those warnings to be reported. By changing `version` to false in Boards, it causes it to load whatever version that the web app provides for a singleton module. I also changed `requiredVersion` to `false` as well since it meant that sometimes we'd end up with two versions of React loaded which caused errors